### ISSUE TITLE
Stackoverflow in logs

### ISSUE THREAD

**Issue by LukasForst** [mild-blue/covid-vaxx#287]
* unhandled -> indicates some shitstorm

my guess is that `ThrowableProxyUtil.asString` does that in some corner cases with causes 
![image](https://user-images.githubusercontent.com/14038818/121167840-981f6580-c852-11eb-9d68-7d7e7ffabe75.png)

see - https://jira.qos.ch/browse/LOGBACK-1454

* callId c3a6173b-2a29-46a5-854a-e379d9fabcdc

```
ERROR Unhandled: POST - /api/patient
```
![image](https://user-images.githubusercontent.com/14038818/121167977-bd13d880-c852-11eb-9045-361e3356be23.png)

and there're lot of them

**Comment by LukasForst** [comment]
Maybe using Guava we can fix this - https://stackoverflow.com/a/9291533/7169288 - they can detect the loops - https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Throwables.java#L297

**Comment by LukasForst** [comment]
![image](https://user-images.githubusercontent.com/14038818/121516983-229ccc00-c9ef-11eb-855e-0ddf11abd452.png)

**Comment by LukasForst** [comment]
fixed in #303 however, http client for some reason creates circular dependency in the exception calls

```
An exception io.ktor.network.sockets.SocketTimeoutException was thrown! - Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] msio.ktor.network.sockets.SocketTimeoutException: Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] msat io.ktor.client.features.HttpTimeoutKt.SocketTimeoutException(HttpTimeout.kt:207)at io.ktor.client.engine.apache.ApacheHttpRequestKt.mapCause(ApacheHttpRequest.kt:62)at io.ktor.client.engine.apache.ApacheHttpRequestKt.sendRequest(ApacheHttpRequest.kt:54)at io.ktor.client.engine.apache.ApacheHttpRequestKt$sendRequest$1.invokeSuspend(ApacheHttpRequest.kt)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)Caused by: io.ktor.network.sockets.SocketTimeoutException: Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] msat io.ktor.client.features.HttpTimeoutKt.SocketTimeoutException(HttpTimeout.kt:207)at io.ktor.client.engine.apache.ApacheHttpRequestKt.mapCause(ApacheHttpRequest.kt:62)at io.ktor.client.engine.apache.ApacheResponseConsumer.failed(ApacheResponseConsumer.kt:76)at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.executionFailed(DefaultClientExchangeHandlerImpl.java:99)at org.apache.http.impl.nio.client.AbstractClientExchangeHandler.failed(AbstractClientExchangeHandler.java:426)at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92)at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39)at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502)at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)at java.base/java.lang.Thread.run(Unknown Source)Suppressed: [CIRCULAR REFERENCE: io.ktor.network.sockets.SocketTimeoutException: Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] ms]Caused by: java.net.SocketTimeoutException: 10,000 milliseconds timeout on connection http-outgoing-14 [ACTIVE]... 11 more
--
exception.stacktrace | java.lang.Exception: An exception io.ktor.network.sockets.SocketTimeoutException was thrown! - Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] msio.ktor.network.sockets.SocketTimeoutException: Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] msat io.ktor.client.features.HttpTimeoutKt.SocketTimeoutException(HttpTimeout.kt:207)at io.ktor.client.engine.apache.ApacheHttpRequestKt.mapCause(ApacheHttpRequest.kt:62)at io.ktor.client.engine.apache.ApacheHttpRequestKt.sendRequest(ApacheHttpRequest.kt:54)at io.ktor.client.engine.apache.ApacheHttpRequestKt$sendRequest$1.invokeSuspend(ApacheHttpRequest.kt)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)Caused by: io.ktor.network.sockets.SocketTimeoutException: Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] msat io.ktor.client.features.HttpTimeoutKt.SocketTimeoutException(HttpTimeout.kt:207)at io.ktor.client.engine.apache.ApacheHttpRequestKt.mapCause(ApacheHttpRequest.kt:62)at io.ktor.client.engine.apache.ApacheResponseConsumer.failed(ApacheResponseConsumer.kt:76)at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.executionFailed(DefaultClientExchangeHandlerImpl.java:99)at org.apache.http.impl.nio.client.AbstractClientExchangeHandler.failed(AbstractClientExchangeHandler.java:426)at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92)at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39)at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502)at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)at java.base/java.lang.Thread.run(Unknown Source)Suppressed: [CIRCULAR REFERENCE: io.ktor.network.sockets.SocketTimeoutException: Socket timeout has expired [url=https://api.uzis.cz/api/v1/pacienti/VyhledatDleJmenoPrijmeniRc/XXXXXXX/YYYYYYYY/ZZZZZZZZZZZ?pcz=000&amp;pracovnikNrzpCislo=MMMMMMMMMMMM, socket_timeout=unknown] ms]Caused by: java.net.SocketTimeoutException: 10,000 milliseconds timeout on connection http-outgoing-14 [ACTIVE]... 11 moreat blue.mild.covid.vaxx.service.IsinValidationService.validatePatient(IsinValidationService.kt:54)at blue.mild.covid.vaxx.service.IsinValidationService$validatePatient$1.invokeSuspend(IsinValidationService.kt)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)at io.ktor.util.pipeline.SuspendFunctionGun.resumeRootWith(SuspendFunctionGun.kt:194)at io.ktor.util.pipeline.SuspendFunctionGun.access$resumeRootWith(SuspendFunctionGun.kt:15)at io.ktor.util.pipeline.SuspendFunctionGun$continuation$1.resumeWith(SuspendFunctionGun.kt:89)at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46)at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)at io.ktor.server.netty.EventLoopGroupProxy$Companion.create$lambda-1$lambda-0(NettyApplicationEngine.kt:251)at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)at java.base/java.lang.Thread.run(Unknown Source)
```
